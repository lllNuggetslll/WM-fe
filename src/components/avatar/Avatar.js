import React from "react";
import { string } from "prop-types";
import styled from "styled-components";
import { mobileBreakPoint } from "./../../constants/breakPoints";

const ImgWrapper = styled.div`
  margin-top: 4px;

  img {
    width: ${props => props.width};
    height: ${props => props.height};
    object-fit: cover;
  }

  @media only screen and (max-width: ${mobileBreakPoint}) {
    margin: 0;
    img {
      width: 100px;
      height: 100px;
    }
  }
`;

const Avatar = ({ img, width, height }) => (
  <ImgWrapper width={width} height={height}>
    <img src={img} alt="listing img" />
  </ImgWrapper>
);

Avatar.propTypes = {
  img: string,
  width: string,
  height: string
};

Avatar.defaultProps = {
  img: "",
  width: "70px",
  height: "70px"
};

export default Avatar;
