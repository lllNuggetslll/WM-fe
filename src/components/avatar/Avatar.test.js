import React from "react";
import { shallow } from "enzyme";
import Avatar from "./Avatar";

describe("Avatar", () => {
  it("should render an avatar", () => {
    const imgSrc = "imgSrc";
    const wrapper = shallow(<Avatar img={imgSrc} />);

    expect(wrapper).toMatchSnapshot();
  });
});
