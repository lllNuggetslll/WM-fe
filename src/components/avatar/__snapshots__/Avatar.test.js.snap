// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render an avatar 1`] = `
<styled.div
  height="70px"
  width="70px"
>
  <img
    alt="listing img"
    src="imgSrc"
  />
</styled.div>
`;
